### PR TITLE
Changed code for usersyncup

### PR DESCRIPTION
**Pull request by pm-shashank-jain**

### CODE REVIEW

File: src/cookieSync.js [hunk 1]
```diff
@@ -18,9 +20,33 @@ function triggerPixel(url) {
...
+      elToAppend.insertBefore(elm, elToAppend.firstChild);
+    }
+  } catch (e) {}
+};
+
+
 function process(response) {
   let result = JSON.parse(response);
-  if (result.status === 'OK' || result.status === 'no_cookie') {
+  if (result.status.toLowerCase() === 'ok' || result.status.toLowerCase() === 'no_cookie') {
```
Comment (pm-harshad-mane) [inline src/cookieSync.js]: move strings to constants

File: src/cookieSync.js [hunk 2]
```diff
@@ -93,7 +119,51 @@ function ajax(url, callback, data, options = {}) {
   }
 }
 
-// Send empty data to receive cookie sync status for all prebid server adapters.
-// In next phase we will read placement id's from query param and will only get cookie sync status of bidders participating in auction
-var data = '{}';
-ajax(ENDPOINT, process, data);
\ No newline at end of file
+function getUrlParam(paramName) {
```
Comment (pm-harshad-mane) [inline src/cookieSync.js]: to read two params from URL using this function, one will need to call this function twice and the same URL processing will be done twice, so change this function to return all URL params in an object, then you can call this function once and check required params from the output.

File: src/cookieSync.js [hunk 3]
```diff
@@ -18,9 +20,33 @@ function triggerPixel(url) {
   img.src = url;
 }
 
+function insertUserSyncIframe(url) {
+  let iframe = document.createElement('iframe');
+  iframe.src = url;
+  insertElement(iframe);
+};
+
+function insertElement(elm, doc, target) {
```
Comment (pm-harshad-mane) [inline src/cookieSync.js]: this function is available in util, use one from util

File: src/cookieSync.js (hunk 2 above)
Comment (pm-harshad-mane) [reply src/cookieSync.js]: Move this function to util

File: src/cookieSync.js [hunk 4]
```diff
@@ -18,9 +20,33 @@ function triggerPixel(url) {
   img.src = url;
 }
 
+function insertUserSyncIframe(url) {
```
Comment (pm-harshad-mane) [inline src/cookieSync.js:39]: a similar function is available in util, use the one from util.
mention height and width as zero
some browsers have default CSS applied to elements which can make a 0x0 iframe visible due to attributes like a border, make sure that this iframe is invisible

File: src/cookieSync.js [hunk 5]
```diff
@@ -18,9 +20,33 @@ function triggerPixel(url) {
...
+  let iframe = document.createElement('iframe');
+  iframe.src = url;
+  insertElement(iframe);
```
Comment (pm-harshad-mane) [inline src/cookieSync.js]: pass other parameters to insertElement , insertElement(elm, doc, 'body') is recommended

File: src/cookieSync.js [hunk 6]
```diff
@@ -93,7 +119,51 @@ function ajax(url, callback, data, options = {}) {
...
+          return sParameterName[1];
+        }
+      }
+    }
+  }
+}
+
+function getBidders() {
+  var bidders = getUrlParam("bidders");
+  if (bidders && bidders.length > 0) {
```
Comment (pm-harshad-mane) [inline src/cookieSync.js]: if bidders param is not passed then the code will not execute any pixel.
We should execute all pixels if bidders param is not passed.

File: src/cookieSync.js [hunk 7]
```diff
@@ -93,7 +119,51 @@ function ajax(url, callback, data, options = {}) {
...
+        "indexExchange",
+        "lifestreet"
+      ];
+    }
+  }
+}
+
+
+var data = JSON.stringify({
+  "uuid": "pubmatic" + (Math.random() * 10000).toFixed(4),
```
Comment (pm-harshad-mane) [inline src/cookieSync.js]: what is the purpose of sending current value of uuid? what if we set it to empty? is it just a cache-buster?

File: src/cookieSync.js [hunk 8]
```diff
@@ -93,7 +119,51 @@ function ajax(url, callback, data, options = {}) {
...
+}
+
+function getBidders() {
+  var bidders = getUrlParam("bidders");
+  if (bidders && bidders.length > 0) {
+    var bidderArray = bidders.split(",");
+    if (bidderArray && bidderArray.length > 0) {
+      return bidderArray;
+    } else {
+      return [
```
Comment (pm-harshad-mane) [inline src/cookieSync.js]: move this array to constants

File: src/cookieSync.js (hunk 7 above)
Comment (pm-shashank-jain) [reply src/cookieSync.js]: after discussion with jaydeep removed it

File: src/cookieSync.js (hunk 1 above)
Comment (pm-shashank-jain) [reply src/cookieSync.js]: Done , created constant for these

File: src/cookieSync.js (hunk 2 above)
Comment (pm-shashank-jain) [reply src/cookieSync.js]: Done

File: src/cookieSync.js (hunk 3 above)
Comment (pm-shashank-jain) [reply src/cookieSync.js]: Done

File: src/cookieSync.js (hunk 4 above)
Comment (pm-shashank-jain) [reply src/cookieSync.js:39]: Done

File: src/cookieSync.js (hunk 5 above)
Comment (pm-shashank-jain) [reply src/cookieSync.js]: Done

File: src/cookieSync.js (hunk 6 above)
Comment (pm-shashank-jain) [reply src/cookieSync.js]: Yeah, there was typo in if condition brackets.
Done

File: src/cookieSync.js (hunk 8 above)
Comment (pm-shashank-jain) [reply src/cookieSync.js]: Done

**Pull request merged**